src/engraving/libmscore/note.cpp

```diff
@@ -800,11 +800,12 @@ int Note::tpc() const
 //   tpcUserName
 //---------------------------------------------------------
 
-String Note::tpcUserName(const int tpc, const int pitch, const bool explicitAccidental)
+String Note::tpcUserName(int tpc, int pitch, bool explicitAccidental, bool full)
 {
     const String pitchStr = mtrc("engraving",
-                                 tpc2name(tpc, NoteSpellingType::STANDARD, NoteCaseType::AUTO, explicitAccidental)
-                                 .replace(u"b", u"♭").replace(u"#", u"♯").toUtf8().constChar());
+                                 tpc2name(tpc, NoteSpellingType::STANDARD, NoteCaseType::AUTO, explicitAccidental, full)
+                                 .replace(u"b", !full ? u"♭" : TConv::userName(AccidentalVal::FLAT, full))
+                                 .replace(u"#", !full ? u"♯" : TConv::userName(AccidentalVal::SHARP, full)));
```

Comment:
I think here you need to do something like:
```c++
String pitchStr = tpc2name(tpc, NoteSpellingType::STANDARD, NoteCaseType::AUTO, explicitAccidental, full);

if (!explicitAccidental) {
    pitchStr.replace(u"b", u"♭");
    pitchStr.replace(u"#", u"♯");
}

pitchStr = mtrc("engraving", pitchStr);
```

Otherwise when `tpc2name()` returns `"double flat"` it become `"dou♭le flat"` or `"douflatle flat"`. Similarly for `"double sharp"` because of the `b` in "double".